midi/midi/decode.py

```diff
@@ -434,13 +492,38 @@ def prepare_file(filepath: str,
         for i in range(1, len(file.tracks)):
             file.tracks[i] = combine_and_clean_tracks([file.tracks[i]])
 
+    # remove empty tracks
+    to_remove = list[int]()
+    for i in range(1, len(file.tracks)):
+        if len(file.tracks[i]) == 0:
+            to_remove.append(i)
+
+    to_remove.sort(reverse=True)
+    for i in to_remove:
+        file.tracks.pop(i)
+
+    if len(file.tracks) <= 1:
+        raise ValueError('empty file - no note messages found')
```

Comment:
Why a single track is an error? Is that the metadata track? Can it not contain notes?

Reply:
yes, yes